Formula/mariadb@10.9.rb

```diff
@@ -36,7 +36,9 @@ class MariadbAT109 < Formula
   keg_only :versioned_formula
 
   # See: https://mariadb.com/kb/en/changes-improvements-in-mariadb-109/
-  deprecate! date: "2023-08-01", because: :unsupported
+  # and https://mariadb.org/about/#maintenance-policy
+  # End-of-life on 2023-08-22. Disable date set to 3 months after.
+  disable! date: "2023-11-22", because: :unsupported
```

Comment:
Updated to use official EOL date from https://mariadb.org/about/#maintenance-policy. Went with +3 month disable date to allow users to get last release around EOL date but also schedule removal given large number of `mariadb` versioned formulae.

Release series | Stable (GA) Date | End-of-life
--- | --- | ---
[10.9](https://mariadb.com/kb/en/changes-improvements-in-mariadb-109/) | 22 Aug 2022 | 22 Aug 2023

Reply:
Switching to `deprecate!` to avoid early deprecation message:
```console
❯ brew info mariadb@10.9
==> mariadb@10.9: stable 10.9.5 (bottled) [keg-only]
Drop-in replacement for MySQL
https://mariadb.org/
Deprecated because it is not supported upstream!
```